### PR TITLE
IGNITE-15108 Integrate with actual data layer in Ignite 3.0

### PR DESCRIPTION
**Pull request by korlov42**

### CODE REVIEW

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/TableScanNode.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    /**
+     * @param ctx Execution context.
+     * @param rowType Output type of the current node.
+     * @param desc Table descriptor this node should scan.
+     * @param parts Partition numbers to scan.
+     * @param filters Optional filter to filter out rows.
+     * @param rowTransformer Optional projection function.
+     * @param requiredColumns Optional set of column of interest.
+     */
+    public TableScanNode(
```
Comment (tledkov-gridgain) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/TableScanNode.java:93]: Please add assertion check for the parts

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/ModifyNode.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    /** */
+    protected final TableDescriptor desc;
+
+    /** */
+    private final TableModify.Operation op;
+
+    /** */
+    private final List<String> cols;
+
+    private final RecordView<Tuple> tableView;
```
Comment (tledkov-gridgain) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/ModifyNode.java:50]: Empty comments?

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/ModifyNode.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+                if (!duplicates.isEmpty()) {
+                    IgniteTypeFactory typeFactory = context().getTypeFactory();
+                    RowHandler.RowFactory<Row> rowFactory = context().rowHandler().factory(
+                        context().getTypeFactory(),
+                        desc.insertRowType(typeFactory)
+                    );
+
+                    throw new IgniteInternalException("Failed to INSERT some keys because they are already in cache. " +
+                        "[tuples=" + duplicates.stream().map(tup -> desc.toRow(context(), tup, rowFactory, null))
```
Comment (tledkov-gridgain) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/ModifyNode.java:205]: please put each stream call on separate line and align ones

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/TableScanNode.java (hunk 1 above)
Comment (korlov42) [reply modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/TableScanNode.java:95]: done

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/ModifyNode.java (hunk 2 above)
Comment (korlov42) [reply modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/ModifyNode.java:51]: done

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/ModifyNode.java (hunk 3 above)
Comment (korlov42) [reply modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/ModifyNode.java]: done

**Pull request merged**